Doc: update tutorial about writting of custom PA modules

pirat89 opened the pull request:
This PR improves and extends the original "module writting tutorial". The goal is to make writting of PA modules for people less tricky and help them understand what they should they when they want to create/add new PA module to extend / customize IPU from RHEL 6 -> 7.

The main readme file (for the tutorial) has been rewritten completely. Number of PA modules (examples / templates) have been extended to cover some common use-cases people reported to us. Each module example contains own readme file providing brief info about the module.

*TODO*
- [x] test functionality of the new templates

pirat89 committed:
Doc: update tutorial for custom modules development

Regarding the feedback, we are updating the current tutorial about
development of custom PA modules to reflect possibly the only use-case
to extend in-place upgrade from RHEL 6 to RHEL 7.

The main readme file for the tutorial is updated to provide a basic
overview about the PA modules and development. The existing cases
are extended by additional README files to provide info about each
module example.

Added several other templates for new modules:
    05_add_pre_upgrade_script
    06_add_post_upgrade_script
    07_module_in_python
    08_install_custom_rpms

mportman12 committed:
Miriam's edits

Made edits for grammar, spelling, and word choice.

pirat89 committed:
Doc: tutorial: fix typos + new graph + ...

- fixed various typos
- added IPU 6 -> 7 process graph
- replace compiled compose section

Rezney commented on doc/module_writing_tutorial/00_create_own_compose/README.md:
````diff
@@ -0,0 +1,24 @@
+```  
+>  Note: src_major_version is the major version of the current system and dst_major_version is the major version to which the system is to be upgraded.
+
+4. Make sure you have installed the _preupgrade-assistant-tools_ package, and run the _preupg-xccdf-compose_ tool as follows:
````
Got this message after issuing `preupg-xccdf-compose`:

`No handlers could be found for logger "preupgrade-assistant-debug"`

pirat89 replied:
this is expected most of the time - especially in case of running on system where the PA is not installed or has not been executed yet. The error can be ignored.

Rezney commented on doc/module_writing_tutorial/00_create_own_compose/README.md:
```diff
@@ -0,0 +1,24 @@
+
+`preupg-xccdf-compose <MODULE_SET_DIR>`
+
+5. With no problems with the module set, the tool should have prepared the module set for the use with the Preupgrade Assistant in a new _<MODULE_SET_DIR>-results_ directory
```
Can we inform users what files/dirs they can expect there?

pirat89 replied:
Ignore this document. Basically it's kind of exaplained in the main readme file, but we do not expect people to execute this one to be honest.

Rezney commented on doc/module_writing_tutorial/00_create_own_compose/README.md:
```diff
@@ -0,0 +1,24 @@
+
+6. Run the Preupgrade Assistant with the --contents option:
+
+`preupg -c <MODULE_SET_DIR>-results/all-xccdf.xml`
```
What do I expect when I run this command?

pirat89 replied:
That the compose will be processed / executed. But this is not actually required from users to do.

Rezney commented on doc/module_writing_tutorial/README.md:
```diff
@@ -1,24 +1,252 @@
+
+### Create a module using the preupg-content-creator utility
+
+The simplest way to create an actor is using the `preupg-content-creator`
```
```suggestion
The simplest way to create a module is using the `preupg-content-creator`
```

pirat89 committed:
Update doc/module_writing_tutorial/README.md

Co-authored-by: Michal Reznik (mreznik) <mreznik@redhat.com>

pirat89 committed:
Doc: tutorial: apply fixes regarding the review

pirat89 committed:
Doc: tutorial: fix the 08_install_custom_rpms template

There was a typo in the code checking the post-upgrade script exists,
so the module ended with the error message everytime. Fixed the
detection of the post-upgrade script file.

Rezney reviewed (approved):
The templates functionality verified. Good to go!

pirat89 commented:
Merging! Just open new PR in case of any additional wanted changes :)
